doc/api/http.md

```diff
@@ -1869,16 +1869,20 @@ added: v0.5.9
 Global instance of `Agent` which is used as the default for all HTTP client
 requests.
 
-## http.request(options[, callback])
+## http.request([url] [, options] [, callback])
```

Comment:
I feel like this should be something like: `url[, options][, callback]` to show that at least one argument is required and not that both are optional with perhaps additional explanation below *or* follow in the footsteps of documented functions like [this](https://nodejs.org/docs/latest/api/net.html#net_net_createconnection) and show multiple signatures for the same function. I'm not sure if we have a standard yet for showing multiple signatures for more complicated situations.

Ditto for `https.request`.

Reply:
I followed the example of [response.end](https://nodejs.org/docs/latest/api/http.html#http_response_end_data_encoding_callback).

I don't think it is a good idea to describe the syntax in a way that implies that url is required even if subsequent text says to ignore that implication.

Technically, nothing is required, as `http.request()` would be evaluated the same as `http.request({ })`.

Reply:
Also, `http.request` accepts a `url` parameter despite ClientRequest giving this argument a different name so as to not collide with the `url` package which was previously required.  Chose the name `input` based on [Request()](https://developer.mozilla.org/en-US/docs/Web/API/Request/Request).

Reply:
FWIW, we have a section with various signatures:

https://nodejs.org/docs/latest/api/console.html#console_new_console_stdout_stderr_ignoreerrors
https://nodejs.org/docs/latest/api/console.html#console_new_console_options